From 9f6d59d620da5cbecdedd0a328ef2f7131f82278 Mon Sep 17 00:00:00 2001
From: Richard Kuo <rkuo@codeaurora.org>
Date: Mon, 17 Oct 2016 19:37:56 -0500
Subject: [PATCH 3/8] Hexagon: change default DMA memory reserved to zero

Not all platforms need DMA memory, and the 16M lastpg alignment was really
just to make the DMA bigpages line up.

Signed-off-by: Richard Kuo <rkuo@codeaurora.org>
---
 arch/hexagon/mm/init.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/hexagon/mm/init.c b/arch/hexagon/mm/init.c
index d02dd56..00bd5de 100644
--- a/arch/hexagon/mm/init.c
+++ b/arch/hexagon/mm/init.c
@@ -1,7 +1,7 @@
 /*
  * Memory subsystem initialization for Hexagon
  *
- * Copyright (c) 2010-2013, The Linux Foundation. All rights reserved.
+ * Copyright (c) 2010-2013,2016, The Linux Foundation. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 and
@@ -158,7 +158,7 @@ void __init paging_init(void)
 }
 
 #ifndef DMA_RESERVE
-#define DMA_RESERVE		(4)
+#define DMA_RESERVE		(0)
 #endif
 
 #define DMA_CHUNKSIZE		(1<<22)
@@ -200,10 +200,11 @@ void __init setup_arch_memory(void)
 	/*  Prior to this, bootmem_lastpg is actually mem size  */
 	bootmem_lastpg += ARCH_PFN_OFFSET;
 
+#if DMA_RESERVE
 	/* Memory size needs to be a multiple of 16M */
 	bootmem_lastpg = PFN_DOWN((bootmem_lastpg << PAGE_SHIFT) &
 		~((BIG_KERNEL_PAGE_SIZE) - 1));
-
+#endif
 	/*
 	 * Reserve the top DMA_RESERVE bytes of RAM for DMA (uncached)
 	 * memory allocation
-- 
2.7.4

